kernel/aster-nix/src/syscall/socket.rs

```diff
@@ -35,7 +35,7 @@ pub fn sys_socket(domain: i32, type_: i32, protocol: i32) -> Result<SyscallRetur
         (
             CSocketAddrFamily::AF_INET,
             SockType::SOCK_DGRAM,
-            Protocol::IPPROTO_IP | Protocol::IPPROTO_UDP,
+            Protocol::IPPROTO_IP | Protocol::IPPROTO_UDP | Protocol::IPPROTO_ICMP,
```

Comment:
`smoltcp` has a seperate [ICMP socket](https://docs.rs/smoltcp/latest/smoltcp/socket/icmp/struct.Socket.html).  Should we add a seperate `IcmpSocket` instead of just using UDP socket?

Reply:
I shouldn't reuse existing existing entry at least it seems. It just because previously I `strace` ping on my PC, where it uses SOCK_DGRAM. But latter when I check in the busybox one, it simply use SOCK_RAW.
At least to make current `ping` works, I should always create a new entry, not sure Icmp or Raw socket though.

Reply:
Well, I don't know the complexity of raw socket.

From my points of view, if the raw socket is not very complex, you can implement raw socket to run `ping` from busybox.

If raw socket is complex to implement, you can copy the `ping` executable from host into Asterinas (perhaps together with other dependent libraries). We can run the host `ping` instead of the `ping` from busybox.